packages/terra-clinical-item-collection/src/Item.jsx

```diff
@@ -0,0 +1,188 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import ItemView from 'terra-clinical-item-view';
+import List from 'terra-list';
+import Table from 'terra-table';
+import 'terra-base/lib/baseStyles';
+import Utils from './_ItemCollectionUtils';
+import styles from './ItemCollection.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The display elements to be presented.
+   */
+  children: PropTypes.node,
+  /**
+   * The start accessory element to be presented.
+   */
+  startAccessory: PropTypes.node,
+  /**
+   * The comment element to be presented.
+   */
+  comment: PropTypes.node,
+  /**
+   * The end accessory element to be presented.
+   */
+  endAccessory: PropTypes.node,
+  /**
+   * When displayed as a list item, the column layout in which to present the displays.
+   */
+  listItemLayout: PropTypes.oneOf(['oneColumn', 'twoColumns']),
+  /**
+   * When displayed as a list item, the text color emphasis when using the two columns layout.
+   */
+  listItemTextEmphasis: PropTypes.oneOf(['default', 'start']),
+  /**
+   * When displayed as a list item, whether or not all text should truncate.
+   */
+  isListItemTruncated: PropTypes.bool,
+  /**
+   * The vertical alignment of the start and end accessories.
+   */
+  accessoryAlignment: PropTypes.oneOf(['alignTop', 'alignCenter']),
+  /**
+   * Wether or not the item is selectable. If true, the item is given list and table hover and focus styles, tabIndex
+   * set to 0, and onClick and onKeyDown callbacks set to the onSelect function provided via Item Collection.
+   */
+  isSelectable: PropTypes.bool,
+  /*
+   * Wether or not the item is selected.
+   */
+  isSelected: PropTypes.bool,
+  /**
+   * The view in which the item should be presented. Options are list or table and this will be set by the Item Collection component.
+   */
+  view: PropTypes.oneOf(['list', 'table']),
+};
+
+const defaultProps = {
+  startAccessory: undefined,
+  comment: undefined,
+  endAccessory: undefined,
+  listItemLayout: 'oneColumn',
+  listItemTextEmphasis: 'default',
+  isListItemTruncated: false,
+  accessoryAlignment: 'alignCenter',
+  isSelectable: false,
+  isSelected: false,
+  view: 'list',
+};
+
+function createSelectableProps(onSelect, itemKey) {
+  const selectableProps = { isSelectable: true };
+  selectableProps.tabIndex = 0;
+
+  if (onSelect) {
+    if (!itemKey) {
+      /* eslint-disable no-console */
+      console.error('Key is required for correct selectable implementation.');
```

Comment:
If they pass an onSelect without an itemKey, do we expect the component to error out, or just ignore the requested implementation as it's invalid? If it's the later, it maybe better to do console.warn inside a non production environment by using the following wrapper

```
if (process.env !== 'production') {
}
```

If it's the first one where we expect an error, should we also do something when onSelect isn't valid as well?

Reply:
Consumers should be providing an 'key' on each item that will be mapped to the 'read-only' itemKey prop if they are implementing a selectable Item Collection. React allows will catch duplicate key values, however a key does not have to be provided or could can be null. 

I added this catch for the instance a consumer did not provide a key because the Item Collection will render as expected with the correct selectable behaviors & styling, however the catch is the itemKeys callback will be undefined. IMO the component should not error out because the lack of a valid key is an implementation issue on the consumer's side.

Reply:
I'm fine with that. In that case, can you switch to a console.warn?